cmd/metacache-walk.go

```diff
@@ -220,16 +228,16 @@ func (s *xlStorage) WalkDir(ctx context.Context, opts WalkDirOptions, wr io.Writ
 				meta.name = pathJoinBuf(&sb, current, meta.name)
 				meta.name = decodeDirObject(meta.name)
 
+				if err := send(meta); err != nil {
```

Comment:
Seems like you are creating the race here, by referencing `meta.metadata` after sending it (you do this below as well). There is no reason for that change.

Reply:
I thought we need to do this after we wrote to channel, looks like it's not possible.

Let me revert it.